### PR TITLE
Change to pkg/util/node.UpdateNodeStatus

### PR DESCRIPTION
**Pull request by dixudx**
**What this PR does / why we need it**:

> // TODO: Change to pkg/util/node.UpdateNodeStatus.

**Which issue this PR fixes** *(optional, in `fixes #<issue number>(, fixes #<issue_number>, ...)` format, will close that issue when PR gets merged)*: fixes #

**Special notes for your reviewer**:
/cc @brendandburns @dchen1107 @lavalamp 

**Release note**:

```release-note
None
```

### CODE REVIEW

**Comment by jsafrane** [comment]
@dixudx, start with

```
W0731 13:43:59.094] # k8s.io/kubernetes/pkg/controller/volume/attachdetach/statusupdater
W0731 13:43:59.095] pkg/controller/volume/attachdetach/statusupdater/node_status_updater.go:22: imported and not used: "encoding/json"
W0731 13:43:59.096] pkg/controller/volume/attachdetach/statusupdater/node_status_updater.go:30: imported and not used: "k8s.io/kubernetes/vendor/k8s.io/apimachinery/pkg/util/strategicpatch"
```

And 
```
I0731 08:01:15.332] Run ./hack/update-bazel.sh
```

**Comment by dixudx** [comment]
@jsafrane Thanks for pointing out. I thought it was those CIs. They are broken frequently these days.

**Comment by dixudx** [comment]
@gnufied @jingxu97 PTAL. Thanks

**Comment by jingxu97** [comment]
@dixudx, sorry that I missed your PR.

I don't think this PR needs a release note. Could you remove it from the commit message?

File: pkg/util/node/node.go [hunk 1]
```diff
@@ -151,9 +151,36 @@ func SetNodeCondition(c clientset.Interface, node types.NodeName, condition v1.N
...
+}
+
+// UpdateNodeStatus updates node status.
+func UpdateNodeStatus(c clientset.Interface, nodeName types.NodeName, oldNode *v1.Node, newNode *v1.Node) (*v1.Node, error) {
+	patchBytes, err := preparePatchBytesforNodeStatus(nodeName, oldNode, newNode)
+	if err != nil {
+		return nil, err
+	}
+
+	updatedNode, err := c.CoreV1().Nodes().PatchStatus(string(nodeName), patchBytes)
```
Comment (jingxu97) [inline pkg/util/node/node.go]: Why you use CoreV1 here?

Actually I feel these two functions PatchNodeStatus and UpdateNodeStatus are the same. We only need one of those.

Comment (dixudx) [reply pkg/util/node/node.go]: Will remove one.

> Why you use CoreV1 here?

`Core` is deprecated. Do we still use it instead of `CoreV1`?

**Comment by dixudx** [comment]
ping @jingxu97 PTAL. Thanks.

File: pkg/util/node/node.go (hunk 1 above)
Comment (jingxu97) [reply pkg/util/node/node.go]: I am not sure about this actually. Maybe we should confirm with someone in api team.

cc @caesarxuchao Should we use CoreV1 or just Core?

Comment (caesarxuchao) [reply pkg/util/node/node.go]: CoreV1 is better.

Comment (jingxu97) [reply pkg/util/node/node.go]: thanks!
@dixudx could you use CoreV1 instead based on @caesarxuchao 's suggestion?

Comment (dixudx) [reply pkg/util/node/node.go]: @jingxu97 @caesarxuchao Done. PTAL. Thanks.

**Comment by dixudx** [comment]
ping @jingxu97 Rebased. PTAL. Thanks.

**Comment by dixudx** [comment]
/assign @dchen1107

ping @dchen1107 Need your `approval`. PTAL. Thanks.

**Comment by dixudx** [comment]
ping @lavalamp @smarterclayton Need your `/approve no-issue`. PTAL. Thanks

File: pkg/controller/volume/attachdetach/statusupdater/node_status_updater.go [hunk 2]
```diff
@@ -99,48 +96,12 @@ func (nsu *nodeStatusUpdater) UpdateNodeStatuses() error {
...
+	if _, err := nodeutil.PatchNodeStatus(nsu.kubeClient.CoreV1(), nodeName, nodeObj, nodeCopy); err != nil {
+		return err
 	}
-	glog.V(4).Infof(
-		"Updating status for node %q succeeded. patchBytes: %q VolumesAttached: %v",
-		nodeName,
-		string(patchBytes),
-		node.Status.VolumesAttached)
 
+	glog.V(4).Infof("Updating status for node %q succeeded. VolumesAttached: %v", nodeName, attachedVolumes)
```
Comment (rootfs) [inline pkg/controller/volume/attachdetach/statusupdater/node_status_updater.go]: `patchBytes` helps capture the entire transaction, get it and print here.

**Comment by dixudx** [comment]
@rootfs Updated. PTAL. Thanks.

**Comment by dixudx** [comment]
@jingxu97 @rootfs Rebased. PTAL. Thanks.

**Comment by dixudx** [comment]
ping @dchen1107 @smarterclayton for approval. Thanks.

**Comment by dixudx** [comment]
ping @thockin  @smarterclayton for approval. Thanks.

**Comment by dixudx** [comment]
ping @brendandburns @smarterclayton Needs your `/approve no-issue`.

**Comment by dixudx** [comment]
ping @thockin PTAL. Needs your approval. Thanks.

**Comment by jingxu97** [comment]
@thockin @dchen1107 could you please help approve this PR? Thanks!

**Pull request merged**